### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by arhamathar**
get and post request api for blog added

### CODE REVIEW

**Comment by Faisal-Manzer** [comment]
Merge all commits to one... and force push
```git reset --soft HEAD~4
git commit -m "New Commit message```

And in commit message explain what you have done.

File: helpers/constants/app.constants.helper.ts [hunk 1]
```diff
@@ -13,4 +13,9 @@ export const NAV_BAR_LINKS: NavBarLinks[] = [
         href: '/admin/users/',
         permission: PERMISSIONS.VIEW_USERS,
     },
+    {
+        label: 'About Us',
+        href: '/about',
+        // permission: PERMISSIONS.VIEW_USERS,
```
Comment (Faisal-Manzer) [inline helpers/constants/app.constants.helper.ts]: remove comment

File: helpers/errors/codes.errors.helper.ts [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
+    'UNKNOWN_ERROR_OCCURRED',
+    500,
+    'Some unknown error occurred while processing request.',
+];
 
-export const INVALID_AMOUNT_ERROR = ['INVALID_AMOUNT_ERROR', 400,];
+export const INVALID_AMOUNT_ERROR = ['INVALID_AMOUNT_ERROR', 400];
 export const INVALID_ORDER_ID = ['INVALID_AMOUNT_ERROR', 200, 'OrderId is not valid'];
+
+export const INVALID_DATA_ERROR = ['INVALID_DATA_ERROR', 422, 'Please enter all fields!'];
```
Comment (Faisal-Manzer) [inline helpers/errors/codes.errors.helper.ts:13]: Error handling should be more specific. For any form give proper error messages on what field failed what.

File: models/blog.models.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import mongoose from 'mongoose';
+
+const Blog = new mongoose.Schema(
+    {
+        title: {
+            type: String,
+            required: true,
+        },
+        description: {
```
Comment (Faisal-Manzer) [inline models/blog.models.ts:9]: Rename this field to markdown or html content... whatever you are save saving.

File: pages/about.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/* eslint-disable jsx-quotes */
```
Comment (Faisal-Manzer) [inline pages/about.tsx]: Remove this. Enable eslint in your editor.

File: pages/api/blog/index.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        //     label: '',
+        // };
+
+        // const isPermitted = hasPermission(session, permission);
+        // console.log(isPermitted);
+
+        // if (!isPermitted) {
+        //     console.log('NO PERMISSION');
+
+        //     return error(__.NO_PERMISSION);
```
Comment (Faisal-Manzer) [inline pages/api/blog/index.ts]: remove unnecessary comments

File: pages/api/blog/index.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+        try {
+            const blogs = await Blog.find({});
+
+            res.status(200).json({ blogs });
+        } catch (err) {
+            error(__.UNKNOWN_ERROR_OCCURRED);
+            console.log(err);
+        }
+    }
```
Comment (Faisal-Manzer) [inline pages/api/blog/index.ts]: for list use helper. don't list like this. it will give all blogs at once.

File: pages/api/blog/index.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+    if (req.method === 'POST') {
+        const { title, description } = req.body;
+
+        if (!title || title.trim() === '' || !description || description.trim() === '') {
+            return error(__.INVALID_DATA_ERROR);
+        }
+
+        let session;
```
Comment (Faisal-Manzer) [inline pages/api/blog/index.ts:22]: you can use session directly. in user is not authenticated it will give null.

File: pages/api/blog/index.ts [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        try {
+            session = await getSession({ req });
+        } catch (e) {
+            error(__.UNKNOWN_ERROR_OCCURRED);
+        }
```
Comment (Faisal-Manzer) [inline pages/api/blog/index.ts:27]: no need for this block

File: pages/api/blog/index.ts [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+        if (!session) {
+            return error(__.LOGIN_REQUIRED);
+        }
+
+        const permission = {
+            code: session.user.permissions[0],
+            label: '',
+        };
```
Comment (Faisal-Manzer) [inline pages/api/blog/index.ts:36]: ??? what is this?

File: pages/api/blog/index.ts [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+            return error(__.NO_PERMISSION);
+        }
+
+        const newBlog = new Blog({
+            title,
+            description,
+            image:
+                'https://images.unsplash.com/photo-1619806629131-959b8fdc50a1?ixid=MnwxMjA3fDB8MHxlZGl0b3JpYWwtZmVlZHwxOHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=500&q=60',
+            author: session.user.uid,
+        });
```
Comment (Faisal-Manzer) [inline pages/api/blog/index.ts]: before saving run proper check on what you are saving... you can use ```yup``` for this.
also give proper error messages for what went wrong.

File: pages/donate.tsx [hunk 11]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable jsx-quotes */
```
Comment (Faisal-Manzer) [inline pages/donate.tsx]: remove this. we are using single quotes every where. use that only.

File: pages/donate.tsx [hunk 12]
```diff
@@ -36,33 +39,25 @@ const Donate: NextPage = () => {
...
-            authenticationRequired={true}
-        >
-            <div className='grid grid-cols-1 md:grid-cols-2 space-y-4 space-x-4'>
+        <Page title="Show some love, Donate us" description="" hideDonateBanner={true} authenticationRequired={true}>
+            <div className="grid grid-cols-1 md:grid-cols-2 space-y-4 space-x-4">
                 <div>
-                    <PageHeading className='text-center'>
-                        Donate <HeartIcon className='h-6 md:h-12 text-yellow-500' />
+                    <PageHeading className="text-center">
+                        Donate <HeartIcon className="h-6 md:h-12 text-yellow-500" />
```
Comment (Faisal-Manzer) [inline pages/donate.tsx]: why double quotes????

**Review by Faisal-Manzer — CHANGES_REQUESTED** [review]

File: pages/api/blog/index.ts (hunk 9 above)
Comment (arhamathar) [reply pages/api/blog/index.ts:36]: It is checking if user has the permission to add blog as not all authenticated user has that permission.

File: pages/api/blog/index.ts (hunk 5 above)
Comment (arhamathar) [reply pages/api/blog/index.ts]: okay it is just for testing purpose

File: pages/api/blog/index.ts (hunk 8 above)
Comment (arhamathar) [reply pages/api/blog/index.ts:27]: so how we will get user related info ??

**Review by Faisal-Manzer — APPROVED** [review]

**Pull request merged**